lib/widgets/compose_box.dart

```diff
@@ -574,7 +586,17 @@ Future<Iterable<_File>> _getFilePickerFiles(BuildContext context, FileType type)
 
   return result.files.map((f) {
     assert(f.readStream != null);  // We passed `withReadStream: true` to pickFiles.
-    return _File(content: f.readStream!, length: f.size, filename: f.name);
+    String? mimeType;
+    if (f.path != null) {
+      mimeType = lookupMimeType(f.path!,
+        headerBytes: f.bytes?.take(defaultMagicNumbersMaxLength).toList());
```

Comment:
Hmm, this upstream API seems a bit quirky: it requires the path, even though it won't actually even look at the path if it finds a magic number in `headerBytes`.

I suspect this is a consequence of someone moving a bit too quickly through this when updating the API for NNBD — that they should have made it explicitly accept `String? path`.

Anyway, if we do find ourselves with no path, we can get better behavior here if we ask this function to look at the magic numbers anyway. To do that, we can say something like `f.path ?? ''` — so the extension is `''`, which should be a value that doesn't map to any particular type so we still get `null` if no magic numbers match.

Reply:
https://github.com/dart-lang/mime/issues/11

Looks like my guess about the NNBD migration being involved was wrong — that issue thread predates NNBD. Anyway, we have a workaround.

Reply:
Thanks for spotting this!